### PR TITLE
chore: Update controller-tools

### PR DESCRIPTION
**Pull request by chrkl**
## Description

Previously used controller-tools 0.11.3 might have issues with latest Golang release.

Changes proposed in this pull request (what was done and why):

- Update controller-tools to 0.14.0
- Updated table-gen
- Improve wording in CRD description

Changes refer to particular issues, PRs or documents:

- https://github.com/kubernetes-sigs/controller-tools/issues/888

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

**Comment by chrkl** [comment]
New controller-gen introduces new-lines in CRD docs, which breaks our generated Markdown tables. Fix in table-gen is required first.

File: config/crd/bases/operator.kyma-project.io_telemetries.yaml [hunk 1]
```diff
@@ -87,9 +91,9 @@ spec:
                           strategy type.
                         properties:
                           static:
-                            description: Static is a scaling strategy allowing you
-                              to define a custom amount of replicas to be used for
-                              the gateway. Present only if Type = StaticScalingStrategyType.
+                            description: |-
+                              Static is a scaling strategy allowing you to define a custom amount of replicas to be used for the gateway. Present only if Type =
```
Comment (NHingerl) [inline config/crd/bases/operator.kyma-project.io_telemetries.yaml:95]:
```suggestion
                              Static is a scaling strategy enabling you to define a custom amount of replicas to be used for the gateway. Present only if Type =
```

File: config/crd/bases/operator.kyma-project.io_telemetries.yaml [hunk 2]
```diff
@@ -112,47 +116,47 @@ spec:
...
+                    \   // other fields\n\t}"
                   properties:
                     lastTransitionTime:
-                      description: lastTransitionTime is the last time the condition
-                        transitioned from one status to another. This should be when
-                        the underlying condition changed.  If that is not known, then
-                        using the time when the API field changed is acceptable.
+                      description: |-
+                        lastTransitionTime is the last time the condition transitioned from one status to another.
+                        This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
```
[the remark this answers is not in the document]
Comment (chrkl) [reply config/crd/bases/operator.kyma-project.io_telemetries.yaml:137]: This description is imported from a [dependency](https://github.com/kubernetes/apimachinery/blob/d82afe1e363acae0e8c0953b1bc230d65fdb50e2/pkg/apis/meta/v1/types.go#L1555) and not under our control.

**Review by NHingerl — APPROVED** [review]

**Review by hisarbalik — APPROVED** [review]

**Pull request merged**